### ISSUE TITLE
How should one make sure the hub is up and running all the time?

### ISSUE THREAD

**Issue by safrrhmn** [groupon/Selenium-Grid-Extras#217]
This is more like a question and query not a bug with this.
I have currently 15 nodes and one hub supporting multiple jobs. Since, all the jobs depend on same hub, we need to make sure the service of hub is seamless at any point of time. This morning, I came in and noticed the hub went down(for various reason: network maintenance etc) and all the jobs on Jenkins failed. So, all the strategies to keep the node clean, restart are in vain.
I wonder, what the community do in such case? Is it possible to have a backup hub which will take responsibility if primary ones go down?

**Comment by smccarthy** [comment]
Is there any way to detect that the hub is down, and auto restart it? I think that solution would be a lot easier than a backup hub

**Comment by safrrhmn** [comment]
@smccarthy  Sure. I have a monitor on. But, that way any jobs running (which are usually 3 or 4 having few hundreds tests) fail. And I have to manually trigger them all and also the history of test lost on Jenkins.

**Comment by smccarthy** [comment]
@safrrhmn Good point. It might be possible to setup a backup hub/grid. What if you handled it in your code? If you try to run a test, and it returns with a failure because the hub is down, programmatically change where you point your test to a backup hub ?

**Comment by safrrhmn** [comment]
@smccarthy I thought about that as well. Again, that way I have to maintain two sets of nodes/hubs. And, as I said it does not happen very often(happened twice in 8 months)
